### PR TITLE
NT-1920: Setup Skeleton Activity and ViewModel for Root Comments

### PR DESCRIPTION
**Pull request by leighdouglas**
# 📲 What

This is the start of feature work on threaded comments. In this PR, we are:
- Deprecating legacy comment classes, tests, v1 api models and layouts
- Creating skeleton classes for `CommentsActivity` and `CommentsViewModel` for root comments. 

# 🤔 Why

We will not be using the old UI and legacy code for root comments as we are migrating to a new UI flow, and therefore these classes are no longer needed. Likewise, we will be moving on from the v1 api calls for comments and shifting to graphQL, and thus creating new models for these queries. 

# 👀 See

These are skeleton classes and have no UI as of yet. 

# 📋 QA

Test that fetching, scrolling, posting and updating comments is working as it should. 

# Story 📖

[NT-1920: Setup Skeleton Activity and ViewModel for Root Comments](https://kickstarter.atlassian.net/browse/NT-1920)

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/activities/UpdateActivity.kt [hunk 1]
```diff
@@ -141,7 +141,7 @@ class UpdateActivity : BaseActivity<UpdateViewModel.ViewModel?>(), KSWebView.Del
     }
 
     private fun startCommentsActivity(update: Update) {
-        val intent = Intent(this, CommentsActivity::class.java)
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/activities/UpdateActivity.kt:144]: Well done! 
I'll add the feature flag check on the `UpdatesActivity` as well, I wasn't aware you can also open comments from the "UpdatesActivity" 👍

Comment (leighdouglas) [reply app/src/main/java/com/kickstarter/ui/activities/UpdateActivity.kt:144]: Thanks! Great callout!

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -50,7 +50,7 @@
             android:name=".ui.activities.EditProfileActivity"
             android:theme="@style/SettingsActivity" />
         <activity
```
Comment (Arkariang) [inline app/src/main/AndroidManifest.xml:52]: I do not see the new Activity declared here, is gonna be done in next Pr's @leighdouglas ?

Comment (leighdouglas) [reply app/src/main/AndroidManifest.xml:52]: Oh good callout, forgot about the manifest 😅 i'll update now 👍🏽

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -46,11 +46,14 @@
...
             android:name=".ui.activities.ChangePasswordActivity"
             android:theme="@style/SettingsActivity" />
+        <activity
```
Comment (Arkariang) [inline app/src/main/AndroidManifest.xml:49]: According to this comment you left on previous PR https://github.com/kickstarter/android-oss/pull/1243#discussion_r632628450 the activity should be called `RootCommentsActivity`

**Review by Arkariang — APPROVED** [review]
Small comment about the Activity name, apart form that all good 👍

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (leighdouglas) [reply app/src/main/AndroidManifest.xml:49]: Oh, iOS hasnt changed their naming yet but i think the idea is that they will do it in the future? I can update the classes to be prefixed with "Root" now if we want to get it over with sooner than later.

**Pull request merged**